Visualization/generate_BackBLAST_heatmap.R

```diff
@@ -0,0 +1,686 @@
+#!/usr/bin/env Rscript
+# generate_BackBLAST_heatmap.R
+# Copyright Lee Bergstrand and Jackson M. Tsuji, 2018
+# Plots a newick treefile and BLAST table together as a phylogenetic tree and heatmap
+# See required R packages below.
+
+#####################################################
+## User variables: #################################
+RUN_COMMAND_LINE <- TRUE # If selected, all user input here is ignored, and terminal-based input is expected instead.
+
+# Set other user variables here
+if (RUN_COMMAND_LINE == FALSE) {
+  # See proper descriptions of variables under parse_command_line_input
+  
+  # Required inputs
+  setwd("/home/jmtsuji/Research_General/Bioinformatics/02_git/BackBLAST_Reciprocal_BLAST/test/")
+  input_phylogenetic_tree_filepath <- "GSB_riboproteins_tree_vs3.treefile"
+  input_blast_table_filename <- "GSB_pathway_vs7_e40_best_hits_MOD3.csv"
+  output_pdf_name <- "test.pdf"
+  
+  # Optional inputs (set to 'NA' to ignore)
+  tree_metadata_filename <- NA
+  tree_decorator_colname <- NA
+  map_genome_names <- NA
+  gene_naming_table_filename <- NA
+  bootstrap_cutoff <- NA
+  root_name <- NA #"Ignavibacterium_album_JCM_16511_NC_017464.1" # Optional; set to NA if you want to use the tree as-is.
+    
+  # To add...
+  # tree_naming_mapping <- ""
+  
+}
+#####################################################
+
+#####################################################
+## Load required packages: ##########################
+library(getopt)
+library(plyr)
+suppressMessages(library(tidyverse))
```

Comment:
Does this suppress error messages? We want those to go through.

Reply:
Load messages for a couple libraries are hard to get rid of and bloat the log files when running the code. I haven't found a better solution but to suppress their load messages during startup.